cansniffer.c

```diff
@@ -709,40 +696,50 @@ void readsettings(char* name, int sockfd){
 	int fd;
 	char fname[30] = SETFNAME;
 	char buf[25] = {0};
-	int i,j;
+	int pos,j;
+	bool done = false;
 
 	strncat(fname, name, 29 - strlen(fname)); 
 	fd = open(fname, O_RDONLY);
     
 	if (fd > 0) {
 		if (!sockfd)
 			printf("reading setting file '%s' ... ", fname);
-
-		for (i=0; i < 2048 ;i++) {
-			if (read(fd, &buf, 24) == 24) {
-				if (buf[5] & 1) {
-					if (is_clr(i, ENABLE)) {
-						do_set(i, ENABLE);
-						if (sockfd)
-							rx_setup(sockfd, i);
+ 
+		while(!done){
+			if (read(fd, &buf, 29) == 29) {
+				unsigned long id = strtoul(&buf[1], (char **)NULL, 16);
+				pos = sniftab_index(id);
+
+				if(pos < 0){
+					if(idx < MAX_SLOTS){
+						pos = idx++;
+						sniftab[pos].current.can_id = id;
+					}
+					else
+						break;
+				}
+				if (buf[10] & 1) {
+					if (is_clr(pos, ENABLE)) {
+						do_set(pos, ENABLE);
 					}
 				}
 				else
-					if (is_set(i, ENABLE)) {
-						do_clr(i, ENABLE);
-						if (sockfd)
-							rx_delete(sockfd, i);
+					if (is_set(pos, ENABLE)) {
+						do_clr(pos, ENABLE);
 					}
 				for (j=7; j>=0 ; j--){
-					sniftab[i].notch.data[j] =
-						(__u8) strtoul(&buf[2*j+7], (char **)NULL, 16) & 0xFF;
-					buf[2*j+7] = 0; /* cut off each time */
+					sniftab[pos].notch.data[j] =
+						(__u8) strtoul(&buf[2*j+12], (char **)NULL, 16) & 0xFF;
```

Comment:
You can drop the casts and the ``& 0xff``.
``strtoul(&buf[2 * j + 12], NULL, 16);``